src/test/groovy/com/pretz/parkingmanager/dto/ParkingStartDTOValidationSpec.groovy

```diff
@@ -0,0 +1,166 @@
+        Set<ConstraintViolation<ParkingStartDTO>> violations = validator.validate(testParkingStartDTO)
+
+        then:
+        Assert.assertTrue(violations.isEmpty())
+    }
+
+    def "validation should fail when vehicle id is null"() {
+
+        given:
+        ParkingStartDTO testParkingStartDTO = testBuilder
+                .vehicleId(null)
+                .build()
+
+        when:
+        Set<ConstraintViolation<ParkingStartDTO>> violations = validator.validate(testParkingStartDTO)
+
+        then:
+        Assert.assertEquals(1, violations.size())
+        ConstraintViolation violation = ++violations.iterator()
+        Assert.assertEquals("must not be null", violation.getMessage())
+        Assert.assertEquals("vehicleId", violation.getPropertyPath().toString())
+    }
+
+    def "validation should fail when vehicle id is inconsistent with basic vehicle id regex"() {
+
+        given:
+        ParkingStartDTO testParkingStartDTO = testBuilder
+                .vehicleId("XR%_#@GE")
+                .build()
+
+        when:
+        Set<ConstraintViolation<ParkingStartDTO>> violations = validator.validate(testParkingStartDTO)
+
+        then:
+        Assert.assertEquals(1, violations.size())
+        ConstraintViolation violation = ++violations.iterator()
+        Assert.assertEquals("must match \"[A-Z]{3}[0-9]{4}\"", violation.getMessage())
+        Assert.assertEquals("vehicleId", violation.getPropertyPath().toString())
+
+    }
+
+    def "validation should pass when parking rate id is 1"() {
+
+        given:
+        ParkingStartDTO testParkingStartDTO = testBuilder
+                .parkingRateId(1)
+                .build()
+
+        when:
+        Set<ConstraintViolation<ParkingStartDTO>> violations = validator.validate(testParkingStartDTO)
+
+        then:
+        Assert.assertEquals(0, violations.size())
+    }
+
+    def "validation should pass when parking rate id is 2"() {
+
+        given:
+        ParkingStartDTO testParkingStartDTO = testBuilder
+                .parkingRateId(1)
+                .build()
+
+        when:
+        Set<ConstraintViolation<ParkingStartDTO>> violations = validator.validate(testParkingStartDTO)
+
+        then:
+        Assert.assertEquals(0, violations.size())
+    }
+
+    def "validation should fail when parking rate id is 0"() {
+
+        given:
+        ParkingStartDTO testParkingStartDTO = testBuilder
+                .parkingRateId(0)
+                .build()
+
+        when:
+        Set<ConstraintViolation<ParkingStartDTO>> violations = validator.validate(testParkingStartDTO)
+
+        then:
+        Assert.assertEquals(1, violations.size())
+        ConstraintViolation violation = ++violations.iterator()
+        Assert.assertEquals("must be greater than or equal to 1", violation.getMessage())
+        Assert.assertEquals("parkingRateId", violation.getPropertyPath().toString())
+    }
+
+    def "validation should fail when parking rate id is bigger than 2"() {
```

Comment:
tak na oko wiekszość tych testów sprawdzających walidacje dałoby się sparametryzować

Reply:
Jest to prawda, z testów walidacyjnych nie jestem zadowolony, powinienem był inaczej je zorganizować bo często się powtarza np. walidacja pól z walutą i jest to tylko niepotrzebna pisanina :)